packages/plugins/live-query/src/index.ts

```diff
@@ -0,0 +1,36 @@
+import { Plugin } from '@envelop/types';
+import type { GraphQLSchema } from 'graphql';
+import { NoLiveMixedWithDeferStreamRule, GraphQLLiveDirective } from '@n1ru4l/graphql-live-query';
+import type { InMemoryLiveQueryStore } from '@n1ru4l/in-memory-live-query-store';
+import { addTypes } from '@graphql-tools/utils';
+
+export type UseLiveQueryOptions = {
+  liveQueryStore: InMemoryLiveQueryStore;
+};
+
+const preparedSchemas = new WeakMap<GraphQLSchema, GraphQLSchema>();
+
+export const useLiveQuery = (opts: UseLiveQueryOptions): Plugin => {
+  return {
+    onExecute: ({ executeFn, setExecuteFn }) => {
+      // @ts-expect-error: execute typings do not include AsyncIterable return right now
+      setExecuteFn(opts.liveQueryStore.makeExecute(executeFn));
+    },
+    onValidate: ({ addValidationRule }) => {
+      addValidationRule(NoLiveMixedWithDeferStreamRule);
+    },
+    onSchemaChange: ({ schema, replaceSchema }) => {
+      let patchedSchema = preparedSchemas.get(schema);
+      if (!patchedSchema) {
+        patchedSchema = addTypes(schema, [GraphQLLiveDirective]);
```

Comment:
Not sure whether adding the directive should be scope of the envelop tbh

Reply:
I think we can expose it as string from the plugin package, so it will be easier to add it to the schema

Reply:
And we can check in onSchemaChange if the schema has it or not